os/net/lwip/src/api/sockets.c

```diff
@@ -1454,7 +1487,7 @@ int lwip_select(int maxfdp1, fd_set *readset, fd_set *writeset, fd_set *exceptse
 			if ((readset && FD_ISSET(i, readset)) || (writeset && FD_ISSET(i, writeset)) || (exceptset && FD_ISSET(i, exceptset))) {
 				struct lwip_sock *sock;
 				SYS_ARCH_PROTECT(lev);
-				sock = tryget_socket(i);
+				sock = tryget_socket(i, getpid());
```

Comment:
There are many `getpid()` in this function. How about declaring one local variable and use that if there is no context switching between every `getpid()`?

Reply:
this looks more simple than your suggested method. because according to you method each time we have to compare local var with getpid() to check whether pid is changed or not. so i found it more simpler.